charts/greptimedb-cluster/templates/_helpers.tpl

```diff
@@ -73,3 +73,12 @@ Generate docker config json
 {{- define "dockerConfigJSON" -}}
 {{- printf "{\"auths\":{\"%s\":{\"username\":\"%s\",\"password\":\"%s\",\"auth\":\"%s\"}}}" .Values.customImageRegistry.registry .Values.customImageRegistry.username .Values.customImageRegistry.password (printf "%s:%s" .Values.customImageRegistry.username .Values.customImageRegistry.password | b64enc) -}}
 {{- end -}}
+
+{{/*
+Validate datanode config
+*/}}
+{{- define "validateDatanodeConfig" -}}
+  {{- if and .Values.datanode.enabled .Values.datanodeGroups }}
+    {{- fail "datanode and datanodeGroups cannot be set at the same time" }}
+  {{- end }}
+{{- end }}
```

Comment:
If `Values.datanode.enabled` and `.Values.datanodeGroups` will following error is return:
```
helm upgrade --install mycluster greptimedb-cluster --values greptimedb-cluster/values.yaml -n default
Error: UPGRADE FAILED: execution error at (greptimedb-cluster/templates/cluster.yaml:1:4): datanode and datanodeGroups cannot be set at the same time
```